dbms/src/TestUtils/FunctionTestUtils.cpp

```diff
@@ -115,39 +115,48 @@ void blockEqual(
     }
 }
 
+std::pair<ExpressionActionsPtr, String> buildFunction(Context & context, const String & func_name, const ColumnsWithTypeAndName & columns, const ColumnNumbers & argument_column_numbers, const TiDB::TiDBCollatorPtr & collator)
+{
+    std::vector<NameAndTypePair> source_columns;
+    for (const auto & column : columns)
+        source_columns.emplace_back(column.name, column.type);
+    DAGExpressionAnalyzer analyzer(source_columns, context);
```

Comment:
can we use `DAGExpressionAnalyzer analyzer({}, context);` directly?

Reply:
No, because this means we don't have source columns, and we can not generate a column at runtime

Reply:
`source columns` are only used by tipb::Expr and it won't be used for `ColumnFunctionExpr `?